server.go

```diff
@@ -188,6 +189,19 @@ func NewFromConfig(logger *logrus.Logger, conf Config) (*Server, error) {
 		logger.SetLevel(logrus.DebugLevel)
 	}
 
+	if conf.MutexProfileFraction > 0 {
```

Comment:
<0 isn't bad config; it's treated as a way to disable the profiling (equivalent to 0).

Reply:
Seems like anything below 0 would be nonsensical to pass. From the docs:

> To turn off profiling entirely, pass rate 0. To just read the current rate, pass rate -1. (For n>1 the details of sampling may change.)

What is the benefit to allowing multiple values to work, where exactly one is sufficient?

Reply:
> What is the benefit to allowing multiple values to work, where exactly one is sufficient?

You'd have to ask the authors of the runtime package! :) All the other functions work this way (and are documented as such). `SetMutexProfileFraction` works the same way, but the documentation is inconsistent with the others and its own implementation. I've submitted a patch to bring it in line with the others: https://go-review.googlesource.com/c/go/+/108175

Since this is just a direct call to runtime.SetMutexProfile fraction and we're both documenting and naming it as such, I'd rather not make our input more restrictive than what the runtime package itself allows - we should be consistent with how that behaves.

Reply:
i don't see why we're bound to reproduce their imprecision, especially when switching domains from code into config.

BlockProfileRate is also a fraction (i realize now i misread the docs and got this wrong in another comment), and does document it as <0 - though they also don't have a mode to read the current rate attached there, which makes the API surface different. CPUProfileRate is a hz concept, so it's also not quite the same.

In any case, our use case is more restrictive than the function itself, because it has the "return previous value" concept. AFAICT there is no meaningful reason for a user to to read the current value, so there is no purpose in passing anything <0.